app/models/batch_connect/session.rb

```diff
@@ -182,14 +205,22 @@ def script(opts = {})
         # hopefully lsf also sets the shell from the shebang
       end
 
-      OodCore::Job::Script.new(
-        {
-          content: script_content(bc_opts),
-          job_name: job_name,
-          workdir: output_root,
-          output_path: output_file
-        }.merge script_opts
+      script_opts = {
```

Comment:
This seems like the wrong method to place this in. This method used to be mostly functional and had one purpose: `Session#script` is defined as a method to build an `OodCore::Job::Script` given a hash of options. Its only state is the `cluster.job_config[:adapter]`, which arguably should be an argument as well.

Now, when ever you get the `script` for the corresponding options hash, the data is written to the staged directory.

Probably more appropriate would be a separate method that accepts `OodCore::Job::Script` instance as an argument and writes the data to the files.